pyannote/audio/pipelines/speaker_diarization.py

```diff
@@ -338,6 +338,10 @@ def iter_waveform_and_mask():
         for i, batch in enumerate(batches, 1):
             waveforms, masks = zip(*filter(lambda b: b[0] is not None, batch))
 
+            if torch.cuda.is_available():
+                waveforms = tuple([x.cuda() for x in waveforms])
```

Comment:
I believe that we can cast waveform somewhere before. Because casting to `cuda` for every element in list seems not efficient. May be we can cast original `file` or `waveform` to `cuda` before splitting it into batch? So we don't need to make multiple casts to `cuda` for every element in batch instead of 1 cast